adafruit_max31856.py

```diff
@@ -0,0 +1,306 @@
+        """
+        return (float(self._read_u8(_MAX31856_CJLF_REG)),
+                float(self._read_u8(_MAX31856_CJHF_REG)))
+
+    @reference_temperature_thresholds.setter
+    def reference_temperature_thresholds(self, val): # pylint: disable=invalid-name
+
+        self._write_u8(_MAX31856_CJLF_REG, int(val[0]))
+        self._write_u8(_MAX31856_CJHF_REG, int(val[1]))
+
+    @property
+    def fault(self):
+        """A dictionary with the status of each fault type where the key is the fault type and the
+        value is a bool if the fault is currently active
+
+        ===================   =================================
+        Key                   Fault type
+        ===================   =================================
+        "cj_range"            Cold junction range fault
+        "tc_range"            Thermocouple range fault
+        "cj_high"             Cold junction high threshold fault
+        "cj_low"              Cold junction low threshold fault
+        "tc_high"             Thermocouple high threshold fault
+        "tc_low"              Thermocouple low threshold fault
+        "voltage"             Over/under voltage fault
+        "open_tc"             Thermocouple open circuit fault
+        ===================   =================================
+
+        """
+        faults = self._read_u8(_MAX31856_SR_REG)
+
+        return {
+            "cj_range": bool(faults & _MAX31856_FAULT_CJRANGE),
+            "tc_range": bool(faults & _MAX31856_FAULT_TCRANGE),
+            "cj_high": bool(faults & _MAX31856_FAULT_CJHIGH),
+            "cj_low": bool(faults & _MAX31856_FAULT_CJLOW),
+            "tc_high": bool(faults & _MAX31856_FAULT_TCHIGH),
+            "tc_low": bool(faults & _MAX31856_FAULT_TCLOW),
+            "voltage": bool(faults & _MAX31856_FAULT_OVUV),
+            "open_tc": bool(faults & _MAX31856_FAULT_OPEN)
+        }
+
+    def _perform_one_shot_measurement(self):
+
+        self._write_u8(_MAX31856_CJTO_REG, 0x0)
+        # read the current value of the first config register
+        conf_reg_0 = self._read_u8(_MAX31856_CR0_REG)
+
+        # and the complement to guarantee the autoconvert bit is unset
+        conf_reg_0 &= ~_MAX31856_CR0_AUTOCONVERT
+        # or the oneshot bit to ensure it is set
+        conf_reg_0 |= _MAX31856_CR0_1SHOT
+
+        # write it back with the new values, prompting the sensor to perform a measurement
+        self._write_u8(_MAX31856_CR0_REG, conf_reg_0)
+
+        # copypasta; original comment below, looks to be waiting for
+        # the reading to finish; presumably we can calculate this based
+        # on the number of cycles that will be taken depending on the
+        # currently selected oversampling
+        sleep(0.250)  # MEME FIX autocalculate based on oversampling
+
+    # these were shamelessly stolen from Tony's MAX31865 driver.
+    # I want to re-factor them to be more similar to the Arduino driver
```

Comment:
Remove these "note to self" comments. This follow up is best done by creating an issue.